src/org/javarosa/core/services/transport/payload/ByteArrayPayload.java

```diff
@@ -72,68 +74,69 @@ public ByteArrayPayload(byte[] payload) {
         this.type = IDataPayload.PAYLOAD_TYPE_XML;
     }
 
-    /* (non-Javadoc)
-     * @see org.javarosa.core.services.transport.IDataPayload#getPayloadStream()
-     */
+    @Override
     public InputStream getPayloadStream() {
-
         return new ByteArrayInputStream(payload);
     }
 
-    /* (non-Javadoc)
-     * @see org.javarosa.core.util.externalizable.Externalizable#readExternal(java.io.DataInputStream, org.javarosa.core.util.externalizable.PrototypeFactory)
-     */
+    @Override
     public void readExternal(DataInputStream in, PrototypeFactory pf)
-            throws IOException, DeserializationException {
+        throws IOException, DeserializationException {
         int length = in.readInt();
-        if(length > 0) {
+        if (length > 0) {
             this.payload = new byte[length];
             in.read(this.payload);
         }
         id = ExtUtil.nullIfEmpty(ExtUtil.readString(in));
     }
 
-    /* (non-Javadoc)
-     * @see org.javarosa.core.util.externalizable.Externalizable#writeExternal(java.io.DataOutputStream)
-     */
+    @Override
     public void writeExternal(DataOutputStream out) throws IOException {
         out.writeInt(payload.length);
-        if(payload.length > 0) {
+        if (payload.length > 0) {
             out.write(payload);
         }
         ExtUtil.writeString(out, ExtUtil.emptyIfNull(id));
     }
 
-    /*
-     * (non-Javadoc)
-     * @see org.javarosa.core.services.transport.IDataPayload#accept(org.javarosa.core.services.transport.IDataPayloadVisitor)
-     */
+    @Override
     public <T> T accept(IDataPayloadVisitor<T> visitor) {
         return visitor.visit(this);
     }
 
-    /*
-     * (non-Javadoc)
-     * @see org.javarosa.core.services.transport.IDataPayload#getPayloadId()
-     */
+    public byte[] getPayloadBytes() {
+        return payload;
+    }
+
+    @Override
     public String getPayloadId() {
         return id;
     }
 
-    /*
-     * (non-Javadoc)
-     * @see org.javarosa.core.services.transport.IDataPayload#getPayloadType()
-     */
+    @Override
     public int getPayloadType() {
         return type;
     }
 
+    @Override
     public long getLength() {
         return payload.length;
     }
 
+    @Override
     public int getTransportId() {
         //TODO: Most messages can include this data
         return -1;
     }
+
+    @Override
+    public String toString() {
+        try {
+            return new String(payload, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            logger.error("The encoding being utilized isn't supported.", e);
+        }
+
+        return null;
```

Comment:
Thanks for the log message.

I’m thinking that a toString method should never return null. Maybe an empty String in this case?

Reply:
Yes, very good catch.